src/script/lua_api/l_object.cpp

```diff
@@ -542,16 +551,128 @@ int ObjectRef::l_get_bone_position(lua_State *L)
 	ServerActiveObject *sao = getobject(ref);
 	if (sao == nullptr)
 		return 0;
-
 	std::string bone = readParam<std::string>(L, 2, "");
+	BoneOverride *override = sao->getBoneOverride(bone);
+	push_v3f(L, override->position.vector);
+	v3f euler_rot;
+	override->rotation.next.toEuler(euler_rot);
+	push_v3f(L, euler_rot * core::RADTODEG);
+	return 2;
+}
 
-	v3f position = v3f(0, 0, 0);
-	v3f rotation = v3f(0, 0, 0);
-	sao->getBonePosition(bone, &position, &rotation);
+// set_bone_override(self, bone, override)
+int ObjectRef::l_set_bone_override(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
+	std::string bone;
+	if (!lua_isnil(L, 2))
+		bone = readParam<std::string>(L, 2);
+	BoneOverride *override = new BoneOverride();
+	if (lua_isnil(L, 3) || lua_isnone(L, 3)) {
+		sao->setBoneOverride(bone, override);
+		return 0;
+	}
+	lua_getfield(L, 3, "position");
+	if (!lua_isnil(L, -1)) {
+		override->position.vector = check_v3f(L, 4);
+		lua_getfield(L, -1, "absolute");
+		override->position.absolute = lua_toboolean(L, 5);
+		lua_pop(L, 1);
+		lua_getfield(L, -1, "interpolation_duration");
+		if (lua_isnumber(L, -1))
+			override->position.interpolation_duration = lua_tonumber(L, -1);
+		lua_pop(L, 2);
+	}
+	lua_getfield(L, 3, "rotation");
+	if (!lua_isnil(L, -1)) {
+		override->rotation.next = core::quaternion(check_v3f(L, -1) * core::DEGTORAD);
+		lua_getfield(L, -1, "absolute");
+		override->rotation.absolute = lua_toboolean(L, -1);
+		lua_pop(L, 1);
+		lua_getfield(L, -1, "interpolation_duration");
+		if (lua_isnumber(L, -1))
+			override->rotation.interpolation_duration = lua_tonumber(L, -1);
+		lua_pop(L, 2);
+	}
+	lua_getfield(L, 3, "scale");
+	if (!lua_isnil(L, -1)) {
+		override->scale.vector = check_v3f(L, -1);
+		lua_getfield(L, -1, "absolute");
+		override->scale.absolute = lua_toboolean(L, -1);
+		lua_pop(L, 1);
+		lua_getfield(L, -1, "interpolation_duration");
+		if (lua_isnumber(L, -1))
+			override->scale.interpolation_duration = lua_tonumber(L, -1);
+		lua_pop(L, 2);
+	}
+	sao->setBoneOverride(bone, override);
+	return 0;
+}
 
-	push_v3f(L, position);
-	push_v3f(L, rotation);
-	return 2;
+void push_bone_override(lua_State *L, BoneOverride *override)
+{
+	lua_newtable(L);
+	lua_newtable(L);
+	push_v3f(L, override->position.vector);
+	lua_setfield(L, -2, "vector");
+	lua_pushnumber(L, override->position.interpolation_duration);
+	lua_setfield(L, -2, "interpolation_duration");
+	lua_pushboolean(L, override->position.absolute);
+	lua_setfield(L, -2, "absolute");
+	lua_setfield(L, -2, "position");
```

Comment:
Suggestion: Move to a Lambda, like above.

Also add new lines to separate each section (i.e. table).